sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md

```diff
@@ -81,23 +137,30 @@ export interface Secret {
 export class SecretClient {
     constructor(endPoint: string, credential: TokenCredential, pipelineOrOptions?: ServiceClientOptions | NewPipelineOptions);
     backupSecret(secretName: string, options?: RequestOptionsBase): Promise<Uint8Array | undefined>;
+    beginDeleteSecret(name: string, options?: SecretPollerOptions): Promise<DeleteSecretPoller>;
+    beginRecoverDeletedSecret(name: string, options?: SecretPollerOptions): Promise<RecoverDeletedSecretPoller>;
     protected readonly credential: TokenCredential;
-    deleteSecret(secretName: string, options?: RequestOptionsBase): Promise<DeletedSecret>;
     static getDefaultPipeline(credential: TokenCredential, pipelineOptions?: NewPipelineOptions): ServiceClientOptions;
     getDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<DeletedSecret>;
     getSecret(secretName: string, options?: GetSecretOptions): Promise<Secret>;
-    listDeletedSecrets(options?: GetSecretsOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
-    listSecrets(options?: GetSecretsOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
-    listSecretVersions(secretName: string, options?: GetSecretsOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
+    listDeletedSecrets(options?: RequestOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
+    listSecrets(options?: RequestOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
+    listSecretVersions(secretName: string, options?: RequestOptions): PagedAsyncIterableIterator<SecretProperties, SecretProperties[]>;
     readonly pipeline: ServiceClientOptions;
     purgeDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<void>;
-    recoverDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<Secret>;
     restoreSecret(secretBundleBackup: Uint8Array, options?: RequestOptionsBase): Promise<Secret>;
     setSecret(secretName: string, value: string, options?: SetSecretOptions): Promise<Secret>;
     updateSecretProperties(secretName: string, secretVersion: string, options?: UpdateSecretOptions): Promise<Secret>;
     readonly vaultEndpoint: string;
 }
 
+// @public
+export interface SecretPollerOptions {
+    intervalInMs?: number;
+    requestOptions?: coreHttp.RequestOptionsBase;
+    resumeFrom?: string;
+}
+
```

Comment:
Both methods only receive the request options as their optional parameters, so I extended it to include the poller options by their side. I had the poller options in "pollerOptions" bag before, but other reviewers didn't like that either.

Reply:
@bterlson what should the shape of the option bags for methods which return LRO pollers look like here?

Reply:
This type is correct based on what's currently checked in. However, @sadasant has a follow-up PR that will flatten requestOptions into the outer options type so that property will go away. LRO-related options will live alongside request related options.